todoapp/codelab/25mar2017/03_activity_detail.md

```diff
@@ -0,0 +1,156 @@
+[\[Paso Anterior\]](02_detail_layout.md)
+
+#DetailActivity.java
+
+Declararemos locationSwitch, LocationManager que nos ayudara a obtener el servicio de location dentro de `DetailActivity.java` asi como 2 contantes que utilizaremos pare diferenciar los diferentes request que haremos en esta actividad.
+``` java
+public class DetailActivity extends AppCompatActivity {
+
+private Switch locationSwitch;
+private LocationManager locationManager;
+private static final int REQUEST_LOCATION_ENABLED = 0;
+private static final int REQUEST_LOCATION_PERMISSION = 1;
+``` 
+Dentro de `onCreate(@Nullable Bundle savedInstanceState)` definimos nuestro LocationManager:
+
+```java 
+protected void onCreate(@Nullable Bundle savedInstanceState) {
+..
+ locationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
+}
+``` 
+
+Ahora dentro del metodo `setViews()` buscaremos la vista a través del id y lo guardaremos en `locationSwitch`
+```java 
+private void setViews() {
+..
+locationSwitch = (Switch) findViewById(R.id.switch_location);
+..
+}
+``` 
+
+Ahora necesitamos declarar la funcionalidad de `locationSwitch` cuando este sea activado o desactivado, por lo que agregaremos un listener a `locationSwitch` dentro del método  `setSwitchListener()`:
+
+```java
+ private void setSwitchListener() {
+ ..
+ locationSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
```

Comment:
what is this?, care to explain anonymous classes? does anybody else explain it before?